hazelcast/src/main/java/com/hazelcast/map/impl/record/RecordFactory.java

```diff
@@ -16,12 +16,14 @@
 
 package com.hazelcast.map.impl.record;
 
+import com.hazelcast.internal.serialization.Data;
+
 /**
  * Factory for creating records. Created for every partition.
  *
  * @param <T> the type of object which is going to be created.
  */
 public interface RecordFactory<T> {
 
-    Record<T> newRecord(Object value);
+    Record<T> newRecord(Data key, Object value);
```

Comment:
Now we have to pass key even we don't use it. Maybe we can have a new method for passing key and
we can call newRecord(Data key, Object value) when we really require it. A change like this can be done for this:

public interface RecordFactory<T> {

    default Record<T> newRecord(Object value) {
        throw new UnsupportedOperationException();
    }

    default Record<T> newRecord(Data key, Object value) {
        return newRecord(value);
    }
}```

Reply:
We call `newRecord` from the `AbsrtractRecordStore#createRecord `. We share the `AbsrtractRecordStore` between all storages including tiered-store and HD/behm store. So, we do have to pass the key because we don't have enough context in that code.

Reply:
this for factories, it should be no impact on `AbsrtractRecordStore#createRecord`

Reply:
I still think that one method is much cleaner, because in the proposed approach we have 2 methods both throwing exception by default. Moreover, the factory callers always use the second method with the `key`. With the proposed approach we just keep the code unchanged for some subclasses, but on the other side get less clean code, and still the caller passes the key.

Reply:
Thanks @petrpleshachkov, i'm fine with your approach.